### PR TITLE
fix(CLI): show on-events scripts in `nango deploy` confirmation message

### PR DESCRIPTION
**Pull request by TBonnin**
Depends on https://github.com/NangoHQ/nango/pull/3067

The CLI should show the added/deleted on-event scripts in the `nango deploy` confirmation message.

Screenshot 2024-11-26 at 15 52 20

How to test:
1. Install local CLI: `nango install -g PATH/TO/REPO/packages/cli`
2. Add/Remove `on-events` scripts in your nango.yaml
3. run `nango deploy dev` and check the confirmation message show the added/removed scripts

### CODE REVIEW

File: packages/shared/lib/services/on-event-scripts.service.ts [hunk 1]
```diff
@@ -84,20 +124,88 @@ export const onEventScriptService = {
...
+            for (const script of provider.scripts) {
+                const key = `${config.id}:${script.name}:${script.event}`;
+
+                const maybeScript = existingMap.get(key);
+                if (maybeScript) {
+                    // Script already exists - it's an update
+                    res.updated.push(maybeScript);
+
+                    // Remove from map to track deletions
+                    existingMap.delete(key);
```
Comment (bodinsamuel) [inline packages/shared/lib/services/on-event-scripts.service.ts]: could named `previousMap` (or something like this) for clarity, I didn't understood why we were removing from existing if it was indeed existing

File: packages/shared/lib/services/on-event-scripts.service.ts [hunk 2]
```diff
@@ -84,20 +124,88 @@ export const onEventScriptService = {
...
+                } else {
+                    // Script doesn't exist - it's new
+                    res.added.push({
+                        configId: config.id,
+                        name: script.name,
+                        version: '0.0.1',
```
Comment (bodinsamuel) [inline packages/shared/lib/services/on-event-scripts.service.ts:197]: Out of curiosity:
Did you talk with Bastien about adding on-events-scripts in the templates, managing their versioning, and being able to enable/disable them via the UI?

**Review by bodinsamuel — APPROVED** [review]

File: packages/shared/lib/services/on-event-scripts.service.ts (hunk 1 above)
Comment (TBonnin) [reply packages/shared/lib/services/on-event-scripts.service.ts]: good idea. will fix

File: packages/shared/lib/services/on-event-scripts.service.ts (hunk 2 above)
Comment (TBonnin) [reply packages/shared/lib/services/on-event-scripts.service.ts:197]: No but that's why I pinged him in a previous PR to mention that there would still be a fair amount of work to support the features you mentioned (pre-built, dasbhoard, etc...) for on-events scripts

**Pull request merged**